gis/gccview/gcc_puller_functions.py

```diff
@@ -593,59 +691,75 @@ def get_layer(mapserver_n, layer_id, schema_name, is_audited, cred = None, con =
         LOGGER.error("Invalid mapserver and/or layer Id")
         return
     #--------------------------------
-    if is_audited and primary_key is None:
-        LOGGER.error("Audited tables should have a primary key.")
-    if not(is_audited) and primary_key is not None:
-        LOGGER.error("Non-audited tables do not use the primary key.")
-    #--------------------------------
     keep_adding = True
     counter = 0
-    
-    while keep_adding == True:
+    #--------------------------------
+    if is_manual:
+        while keep_adding == True:
         
-        if counter == 0:
-            return_json = get_data(mapserver, layer_id)
-            if is_audited:
-                (insert_column, excluded_column) = create_audited_table(output_table, return_json, schema_name, primary_key, con)
-            else:
-                (insert_column, schema_parent_table_insert) = create_partitioned_table(output_table, return_json, schema_name, con)
-            
-            features = return_json['features']
-            record_max=(len(features))
-            max_number = record_max
-            
-            if is_audited:
-                insert_audited_data(output_table, insert_column, return_json, schema_name, con)
+            if counter == 0:
+                return_json = get_data(mapserver, layer_id)
+                insert_column = create_table(output_table, return_json, schema_name, con)
+        
+                features = return_json['features']
+                record_max=len(features)
+                max_number = record_max
             else:
-                insert_partitioned_data(schema_parent_table_insert, insert_column, return_json, schema_name, con)
-            
+                return_json = get_data(mapserver, layer_id, max_number = max_number, record_max = record_max)
+        
+            # Insert data into the table
+            insert_data(output_table, insert_column, return_json, schema_name, con)
+        
+            # Update loop variables
             counter += 1
             keep_adding = find_limit(return_json)
-            if keep_adding == False:
+        
+            if keep_adding:
+                max_number += record_max
+            else:
                 LOGGER.info('All records from [mapserver: %s, layerID: %d] have been inserted into %s', mapserver, layer_id, output_table)
-        else:
-            return_json = get_data(mapserver, layer_id, max_number = max_number, record_max = record_max)
+    else:
+        if is_audited and primary_key is None:
+            LOGGER.error("Audited tables should have a primary key.")
+        if not(is_audited) and primary_key is not None:
+            LOGGER.error("Non-audited tables do not use the primary key.")
+        #--------------------------------
+        while keep_adding == True:
```

Comment:
This while loop is very similar to above, I think it can be combined for better readability